lib/Alchemy/Phrasea/Controller/Prod/ToolsController.php

```diff
@@ -165,6 +166,8 @@ public function imageAction(Request $request)
 
             if (!$substituted || $force) {
                 $record->rebuild_subdefs();
+
+                $this->dispatch(RecordEvents::SUB_DEFINITION_REBUILD, new SubDefinitionRebuildEvent($record));
```

Comment:
We should not do rebuild() AND send event (even if rebuild() is only raising a status-bit for a task). We should send event TO rebuild. 
2 options to send events :
- high level, into controllers. Each time a code did call "rebuild_subdefs()" (7 calls in ph), it now sends an event. The "rebuild_subdefs()" is removed.
- low level, into "rebuild_subdefs()". Send the event from here.

Each case the hard code (building subdefs) is moved to a listener. The default listener will simply raise the sb, but a plugin might override the listener and run a worker.

I prefer option 1 since it allows to remove code from the core.